<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for https://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pycqa/pydocstyle.git.
[INFO] Initializing environment for https://github.com/psf/black.
[ERROR] The hook `black` requires pre-commit version 2.9.2 but version 2.4.0 is installed.  Perhaps run `pip install --upgrade pre-commit`.
```
</details>

--- a/.aicoe-ci.yaml
+++ b/.aicoe-ci.yaml
@@ -1,5 +1,4 @@
 check:
-  - thoth-precommit
   - thoth-build
 build:
   base-image: quay.io/thoth-station/s2i-thoth-ubi8-py38:v0.35.0

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,7 +1,7 @@
 ---
 repos:
   - repo: https://github.com/Lucas-C/pre-commit-hooks
-    rev: v1.3.0
+    rev: v1.3.1
     hooks:
       - id: remove-tabs
 
@@ -36,7 +36,7 @@ repos:
       - id: trailing-whitespace
 
   - repo: https://github.com/psf/black
-    rev: 22.6.0
+    rev: 22.8.0
     hooks:
       - id: black
 
@@ -47,3 +47,12 @@ repos:
         additional_dependencies: ['pep8-naming']
         # Ignore all format-related checks as Black takes care of those.
         args: ['--ignore', 'E2,W5', '--select', 'E,W,F,N', '--max-line-length=120']
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.971
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - attrs~=19.3
+          - types-all
+          - pydantic

--- a/.prow.yaml
+++ b/.prow.yaml
@@ -6,7 +6,7 @@ presubmits:
     context: aicoe-ci/prow/pre-commit
     spec:
       containers:
-        - image: quay.io/thoth-station/thoth-precommit-py38:v0.14.3
+        - image: quay.io/thoth-station/thoth-precommit-py38:v0.15.0
           command:
             - "pre-commit"
             - "run"
@@ -17,26 +17,4 @@ presubmits:
               cpu: "300m"
             limits:
               memory: "1Gi"
-              cpu: "500m"
-  - name: thoth-mypy-py38
-    decorate: true
-    skip_report: false
-    always_run: true
-    context: aicoe-ci/prow/mypy
-    spec:
-      containers:
-        - image: quay.io/thoth-station/thoth-pytest-py38:v0.12.6
-          command:
-            - "run-mypy"
-          args:
-            - "."
-            - "--config-file"
-            - "mypy.ini"
-            - "--ignore-missing-imports"
-          resources:
-            requests:
-              memory: "1Gi"
-              cpu: "300m"
-            limits:
-              memory: "2Gi"
               cpu: "500m"

--- a/.thoth.yaml
+++ b/.thoth.yaml
@@ -17,12 +17,6 @@ managers:
   - name: info
   - name: version
     configuration:
-      maintainers:
-        - sesheta
-        - goern
-        - fridex
-        - pacospace
-        - harshad16
       assignees:
         - sesheta
       labels: [bot]

--- a/OWNERS
+++ b/OWNERS
@@ -1,10 +1,16 @@
 # See the OWNERS docs at https://go.k8s.io/owners
-
 approvers:
-  - fridex
-  - pacospace
+  - goern
+  - harshad16
+  - kpostoffice
+  - vannten
   - sesheta
+
 reviewers:
   - kpostoffice
   - harshad16
-  - saisankargochhayat
+  - vannten
+
+emeritus_approvers:
+  - fridex
+  - pacospace

--- a/mypy.ini
+++ b/mypy.ini
@@ -1,2 +1,0 @@
-[mypy]
-plugins = pydantic.mypy

--- a/producer.py
+++ b/producer.py
@@ -61,9 +61,7 @@ p = producer.create_producer()
 
 _GRAPH_DB = GraphDatabase()
 _GRAPH_DB.connect()
-_COUNT = (
-    int(os.getenv("THOTH_GRAPH_REFRESH_COUNT", GraphDatabase.DEFAULT_COUNT)) or None
-)
+_COUNT = int(os.getenv("THOTH_GRAPH_REFRESH_COUNT", GraphDatabase.DEFAULT_COUNT)) or None
 
 _OPENSHIFT = OpenShift()
 
@@ -75,13 +73,9 @@ _THOTH_METRICS_PUSHGATEWAY_URL = os.getenv("PROMETHEUS_PUSHGATEWAY_URL")
 THOTH_GRAPH_REFRESH_SOLVER = int(os.getenv("THOTH_GRAPH_REFRESH_SOLVER", 1))
 _LOGGER.info("Schedule Solver Messages set to - %r", THOTH_GRAPH_REFRESH_SOLVER)
 THOTH_GRAPH_REFRESH_REVSOLVER = int(os.getenv("THOTH_GRAPH_REFRESH_REVSOLVER", 1))
-_LOGGER.info(
-    "Schedule Reverse Solver Messages set to - %r", THOTH_GRAPH_REFRESH_REVSOLVER
-)
+_LOGGER.info("Schedule Reverse Solver Messages set to - %r", THOTH_GRAPH_REFRESH_REVSOLVER)
 THOTH_GRAPH_REFRESH_SECURITY = int(os.getenv("THOTH_GRAPH_REFRESH_SECURITY", 1))
-_LOGGER.info(
-    "Schedule Unanalyzed SI Messages set to - %r", THOTH_GRAPH_REFRESH_SECURITY
-)
+_LOGGER.info("Schedule Unanalyzed SI Messages set to - %r", THOTH_GRAPH_REFRESH_SECURITY)
 COMPONENT_NAME = "graph-refresh-job"
 
 # Metrics Exporter Metrics
@@ -121,11 +115,7 @@ def _unsolved_packages(packages: list) -> list:
     for solver_name in solver_names:
         _LOGGER.info("Checking unsolved packages for solver %r", solver_name)
         solver_info = _OPENSHIFT.parse_python_solver_name(solver_name)
-        for (
-            package_name,
-            version,
-            index_url,
-        ) in _GRAPH_DB.get_unsolved_python_package_versions_all(
+        for (package_name, version, index_url,) in _GRAPH_DB.get_unsolved_python_package_versions_all(
             os_name=solver_info["os_name"],
             os_version=solver_info["os_version"],
             python_version=solver_info["python_version"],
@@ -169,9 +159,7 @@ def main() -> None:
         return
 
     else:
-        _LOGGER.info(
-            "Graph refresh will produce at most %d messages ", max_messages_sent
-        )
+        _LOGGER.info("Graph refresh will produce at most %d messages ", max_messages_sent)
 
     packages: list = []
 
@@ -302,9 +290,7 @@ def main() -> None:
 
     if _THOTH_METRICS_PUSHGATEWAY_URL:
         try:
-            _LOGGER.debug(
-                f"Submitting metrics to Prometheus pushgateway {_THOTH_METRICS_PUSHGATEWAY_URL}"
-            )
+            _LOGGER.debug(f"Submitting metrics to Prometheus pushgateway {_THOTH_METRICS_PUSHGATEWAY_URL}")
             push_to_gateway(
                 _THOTH_METRICS_PUSHGATEWAY_URL,
                 job="graph-refresh",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,0 +1,20 @@
+[tool.black]
+line-length = 120
+target-version = ['py38']
+include = '\.pyi?$'
+exclude = '''
+/(
+    \.git
+  | \.hg
+  | \.mypy_cache
+  | \.tox
+  | \.venv
+  | _build
+  | build
+  | dist
+)/
+'''
+
+[tool.mypy]
+warn_return_any = true
+plugins = ["pydantic.mypy"]

